src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -431,6 +431,17 @@ private static long ParseTimeValue(JToken jToken, string claimName)
             throw LogHelper.LogExceptionMessage(new FormatException(LogHelper.FormatInvariant(LogMessages.IDX14300, LogHelper.MarkAsNonPII(claimName), jToken.ToString(), LogHelper.MarkAsNonPII(typeof(long)))));
         }
 
+        internal static string PrepareSecurityArtifact(string securityArtifact)
+        {
+            if (IdentityModelEventSource.ShowPII)
```

Comment:
>ShowPII [](http://example.com/codeflow?start=41&length=7)

Isn't this the other way around? ShowPII = true means we dont modify the security artifact, otherwise we remove the signature. 

If you agree we should negate this if statement.

Reply:
I think part of the reason for confusion here is there's a lot of conflation of logic going on. This is set to only execute if PII is enabled because that's the only time the string will be logged (otherwise it's scrubbed). This can lead to difficult to debug problems down the line though. Also why I was favoring/suggesting following the pattern of using the decoration pattern (e.g. MarkAsNonPII) so the logic to scrub log lines is consolidated and not spread out which will make it easier to understand.

Reply:
I completely agree on the decorator pattern, not using it will most likely require additional work to adapt to the other helper methods. 

If I understood your explanation we only cleanup the string when show PII is enabled so we we never leak the signature in the logs? If that is the case we shouldnt even have two execution flows and always strip the value out.

This method belongs to the LogHelper as well.